### DISCUSSION TITLE
Google Cloud Vision API integration in the future?

### DISCUSSION THREAD

**Discussion by mindsailor** [theovercomer8/captionr#6] [Ideas]
Would love to add this type of data even if it meant spending on API. 

Below are a few of the bullet points it can handle accord to GPT.

**_**Image classification: It can classify an image into thousands of predefined categories, such as landscapes, animals, food, and more.

Object detection and tracking: It can detect and track multiple objects within an image, including people, buildings, vehicles, and more.

OCR (Optical Character Recognition): It can extract text from images, including handwritten and machine-printed text.

Face detection and recognition: It can detect faces in an image and identify individual faces.

Landmark detection: It can detect and identify over a thousand landmarks, such as the Eiffel Tower or the Empire State Building.

Logo detection: It can detect logos in an image and identify the company or brand associated with the logo.

Label detection: It can detect objects and entities within an image, such as dogs, cats, books, and more.

Explicit content detection: It can detect explicit content, such as adult and violent content, within an image.

Image attributes: It can extract image attributes, such as image properties, such as dominant colors and image quality.

Sentiment analysis: It can detect the sentiment of people in an image, such as happy, sad, angry, or neutral.

SafeSearch: It can filter images based on their perceived level of adult content and violence.

Image cropping and resizing: It can automatically crop an image to focus on the most visually relevant region and resize it to a specified size.**_**

**Comment by theovercomer8** [comment]
Yea, I think thats a feature I'll implement soon.

**Reply by mindsailor** [reply]
Sweet!  Definitely excited at the prospect of being able to leverage this.
